src/cpp/rtps/builtin/discovery/database/DiscoveryDataBase.cpp

```diff
@@ -35,9 +35,79 @@ DiscoveryDataBase::DiscoveryDataBase(
     : server_guid_prefix_(server_guid_prefix)
     , server_acked_by_all_((servers.size() > 0) ? false : true)
     , servers_(servers)
+    , enabled_(true)
 {
 }
 
+DiscoveryDataBase::~DiscoveryDataBase()
+{
+    if (!clear().empty())
+    {
+        logError(DISCOVERY_DATABASE, "Destroying a NOT cleared database");
+    }
+}
+
+std::vector<fastrtps::rtps::CacheChange_t*> DiscoveryDataBase::clear()
+{
+    std::unique_lock<share_mutex_t> lock(sh_mtx_);
+
+    // Cannot clear an enabled database, since there could be inconsistencies after the process
+    if (enabled_)
+    {
+        logError(DISCOVERY_DATABASE, "Cannot clear an enabled database");
+        return std::vector<fastrtps::rtps::CacheChange_t*>({});
+    }
+    logInfo(DISCOVERY_DATABASE, "Clearing DiscoveryDataBase");
```

Comment:
`enabled_` is atomic, so there is no need of locking before checking
```suggestion
    // Cannot clear an enabled database, since there could be inconsistencies after the process
    if (enabled_)
    {
        logError(DISCOVERY_DATABASE, "Cannot clear an enabled database");
        return std::vector<fastrtps::rtps::CacheChange_t*>({});
    }
    
    std::unique_lock<share_mutex_t> lock(sh_mtx_);
    logInfo(DISCOVERY_DATABASE, "Clearing DiscoveryDataBase");
```

Reply:
:+1: